pkg/kapp/preflight/check.go

```diff
@@ -5,26 +5,35 @@ package preflight
 
 import (
 	"context"
+
 	ctldgraph "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/diffgraph"
 )
 
-type CheckFunc func(context.Context, *ctldgraph.ChangeGraph) error
+// The following is the interface for Preflight checks
+type CheckConfig map[string]any
 
 type Check interface {
 	Enabled() bool
 	SetEnabled(bool)
+	SetConfig(CheckConfig) error
 	Run(context.Context, *ctldgraph.ChangeGraph) error
 }
 
+// The following is an example/test/mock Preflight check
+type setFunc func(CheckConfig) error
+type checkFunc func(context.Context, *ctldgraph.ChangeGraph) error
+
 type checkImpl struct {
```

Comment:
I think maybe there is a misunderstanding of what this is for. This is not created solely for mocking and testing purposes but a general utility for wrapping a function as a `preflight.Check`. In hindsight, maybe this was unnecessary for me to include as at the time there was no explicit use for it other than to use with testing, but at the time I could envision this being used similarly to something like https://pkg.go.dev/net/http#HandlerFunc

I don't think the implementation for this should have changed from the previous commit as I don't see how someone could create a check in this way that supports configuration. I'm open to discussing the utility of this in general, but I think any significant changes like removing this all together is something that is out of scope for this PR and ideally would be pushed further upstream.

Reply:
I can get rid of the comment. I _had_ considered putting `Check` into a separate mock file, but felt that went too far. It's not clear that you had intended this to be a general interface for a preflight check.

So, the reason why I have a separate `SetConfig()` function is to allow configuration to be checked before it's used; in other words, to be able to error out early. I was also not too keen on passing the configuration to the `CheckFunc` as it complicates the interface. Admittedly, without saving the configuration somewhere in a generic way fashion, it makes it hard to use a generic `Check` type.

Now that I understand that this wasn't just a _mock_ structure, I can make some tweaks.